### ISSUE TITLE
Running e2e testing with Kata CI

### ISSUE THREAD

**Issue by sboeuf** [cri-o/cri-o#2114]
Kata currently has a nightly build running `sonobuoy` k8s e2e tests. It runs with containerd by default, and it'd be great to have those same tests running for Kata + CRI-O too.

@chavafg ran the same tests a few weeks ago with CRI-O, but he got some errors...
So the first question is, what are the k8s e2e tests that CRI-O usually run with runc? If they are not exactly the same set of tests, then it might explain the difference, but if they are identical, we need to investigate what's is going on before we can run them.

@chavafg could you post the logs you got when running those tests with CRI-O please? And please highlights the ones failing :)

@mrunalp how should we proceed once we can make those tests passing? Should we hook a new Jenkins job dedicated to this, running on every PR?

**Comment by chavafg** [comment]
Hi, sorry for my late response. These are the tests that have been failing when running with cri-o + kata. 

```
[k8s.io] InitContainer [NodeConformance]
  should invoke init containers on a RestartAlways pod [Conformance] [It]

[sig-storage] Projected downwardAPI 
  should provide container's cpu request [NodeConformance] [Conformance]

[sig-storage] Downward API volume
  should provide container's memory limit [NodeConformance] [Conformance] [It]

[sig-node] Downward API
  should provide pod name, namespace and IP address as env vars [NodeConformance] [Conformance] [It]

[sig-storage] Downward API volume
  should provide container's cpu limit [NodeConformance] [Conformance] [It]

[sig-storage] Projected downwardAPI
  should provide container's memory request [NodeConformance] [Conformance] [It]

[sig-storage] Downward API volume
  should provide container's memory request [NodeConformance] [Conformance] [It]

[k8s.io] InitContainer [NodeConformance]
  should not start app containers and fail the pod if init containers fail on a RestartNever pod [Conformance] [It]

[sig-node] Downward API
  should provide container's limits.cpu/memory and requests.cpu/memory as env vars [NodeConformance] [Conformance] [It]

[sig-node] Downward API
  should provide host IP as an env var [NodeConformance] [Conformance] [It]

[sig-storage] Downward API volume
  should provide container's cpu request [NodeConformance] [Conformance] [It]

[sig-node] Downward API
  should provide pod UID as env vars [NodeConformance] [Conformance] [It]

[sig-storage] Projected downwardAPI
  should provide container's memory limit [NodeConformance] [Conformance] [It]

```

All tests fail with the same grpc error: 
```
Apr  2 21:40:43.488: INFO: At 2019-04-02 21:35:43 +0000 UTC - event for pod-init-44caaa71-558f-11e9-88a2-0a580af4004a: {default-scheduler } Scheduled: Successfully assigned e2e-tests-init-container-rhvrl/pod-init-44caaa71-558f-11e9-88a2-0a580af4004a to crio-k8s
Apr  2 21:40:43.488: INFO: At 2019-04-02 21:36:04 +0000 UTC - event for pod-init-44caaa71-558f-11e9-88a2-0a580af4004a: {kubelet crio-k8s} FailedCreatePodSandBox: Failed create pod sandbox: rpc error: code = Unknown desc = container create failed: Failed to check if grpc server is working: rpc error: code = Unavailable desc = transport is closing

Apr  2 21:40:43.488: INFO: At 2019-04-02 21:38:10 +0000 UTC - event for pod-init-44caaa71-558f-11e9-88a2-0a580af4004a: {kubelet crio-k8s} FailedCreatePodSandBox: Failed create pod sandbox: rpc error: code = Unknown desc = error reading container (probably exited) json message: EOF

```

From kata logs: 
```
Apr 02 21:36:19 crio-k8s kata-runtime[3888]: time="2019-04-02T21:36:19.197580458Z" level=info msg="{\"QMP\": {\"version\": {\"qemu\": {\"micro\": 0, \"minor\": 11, \"major\": 2}, \"package\": \"\"}, \"capabilities\": []}}" arch=amd64 command=create container=d7f40e2e89032e5121af6496da439c11228e6f52d214bf1f5dc1f581ce811539 name=kata-runtime pid=3888 source=virtcontainers subsystem=qmp
Apr 02 21:36:19 crio-k8s kata-runtime[3888]: time="2019-04-02T21:36:19.198213261Z" level=info msg="QMP details" arch=amd64 command=create container=d7f40e2e89032e5121af6496da439c11228e6f52d214bf1f5dc1f581ce811539 name=kata-runtime pid=3888 qmp-capabilities= qmp-major-version=2 qmp-micro-version=0 qmp-minor-version=11 source=virtcontainers subsystem=qemu
Apr 02 21:36:19 crio-k8s kata-runtime[3888]: time="2019-04-02T21:36:19.198333161Z" level=info msg="{\"execute\":\"qmp_capabilities\"}" arch=amd64 command=create container=d7f40e2e89032e5121af6496da439c11228e6f52d214bf1f5dc1f581ce811539 name=kata-runtime pid=3888 source=virtcontainers subsystem=qmp
Apr 02 21:36:19 crio-k8s kata-runtime[3888]: time="2019-04-02T21:36:19.198881064Z" level=info msg="{\"return\": {}}" arch=amd64 command=create container=d7f40e2e89032e5121af6496da439c11228e6f52d214bf1f5dc1f581ce811539 name=kata-runtime pid=3888 source=virtcontainers subsystem=qmp
Apr 02 21:36:19 crio-k8s kata-runtime[3888]: time="2019-04-02T21:36:19.199293566Z" level=info msg="VM started" arch=amd64 command=create container=d7f40e2e89032e5121af6496da439c11228e6f52d214bf1f5dc1f581ce811539 name=kata-runtime pid=3888 sandbox=d7f40e2e89032e5121af6496da439c11228e6f52d214bf1f5dc1f581ce811539 source=virtcontainers subsystem=sandbox
Apr 02 21:36:19 crio-k8s kata-runtime[3888]: time="2019-04-02T21:36:19.199386266Z" level=debug msg="Starting regular Kata proxy rather than built-in" arch=amd64 command=create container=d7f40e2e89032e5121af6496da439c11228e6f52d214bf1f5dc1f581ce811539 name=kata-runtime pid=3888 sandbox=d7f40e2e89032e5121af6496da439c11228e6f52d214bf1f5dc1f581ce811539 source=virtcontainers subsystem=kata_agent
Apr 02 21:36:19 crio-k8s kata-runtime[3888]: time="2019-04-02T21:36:19.20022147Z" level=info msg="proxy started" arch=amd64 command=create container=d7f40e2e89032e5121af6496da439c11228e6f52d214bf1f5dc1f581ce811539 name=kata-runtime pid=3888 proxy-pid=3934 proxy-url="unix:///run/vc/sbs/d7f40e2e89032e5121af6496da439c11228e6f52d214bf1f5dc1f581ce811539/proxy.sock" sandbox=d7f40e2e89032e5121af6496da439c11228e6f52d214bf1f5dc1f581ce811539 source=virtcontainers subsystem=kata_agent
Apr 02 21:36:19 crio-k8s kata-runtime[3888]: time="2019-04-02T21:36:19.200594471Z" level=info msg="New client" arch=amd64 command=create container=d7f40e2e89032e5121af6496da439c11228e6f52d214bf1f5dc1f581ce811539 name=kata-runtime pid=3888 source=virtcontainers subsystem=kata_agent url="unix:///run/vc/sbs/d7f40e2e89032e5121af6496da439c11228e6f52d214bf1f5dc1f581ce811539/proxy.sock"
Apr 02 21:36:19 crio-k8s kata-runtime[3888]: time="2019-04-02T21:36:19.597558753Z" level=debug msg="sending request" arch=amd64 command=create container=d7f40e2e89032e5121af6496da439c11228e6f52d214bf1f5dc1f581ce811539 name=grpc.CheckRequest pid=3888 req= source=virtcontainers subsystem=kata_agent
Apr 02 21:36:20 crio-k8s kata-runtime[3888]: time="2019-04-02T21:36:20.894613976Z" level=debug msg="Deleting sandbox cgroup" arch=amd64 command=create container=d7f40e2e89032e5121af6496da439c11228e6f52d214bf1f5dc1f581ce811539 name=kata-runtime pid=3888 sandbox=d7f40e2e89032e5121af6496da439c11228e6f52d214bf1f5dc1f581ce811539 source=virtcontainers subsystem=sandbox
Apr 02 21:36:20 crio-k8s kata-runtime[3888]: time="2019-04-02T21:36:20.894695676Z" level=debug msg="Deleting no constraints cgroup" arch=amd64 command=create container=d7f40e2e89032e5121af6496da439c11228e6f52d214bf1f5dc1f581ce811539 name=kata-runtime path=/kata pid=3888 sandbox=d7f40e2e89032e5121af6496da439c11228e6f52d214bf1f5dc1f581ce811539 source=virtcontainers subsystem=sandbox
Apr 02 21:36:20 crio-k8s kata-runtime[3888]: time="2019-04-02T21:36:20.895094278Z" level=info msg="cleanup agent" arch=amd64 command=create container=d7f40e2e89032e5121af6496da439c11228e6f52d214bf1f5dc1f581ce811539 name=kata-runtime path=/run/kata-containers/shared/sandboxes/d7f40e2e89032e5121af6496da439c11228e6f52d214bf1f5dc1f581ce811539 pid=3888 source=virtcontainers subsystem=kata_agent
Apr 02 21:36:20 crio-k8s kata-runtime[3888]: time="2019-04-02T21:36:20.895164178Z" level=debug msg="Deleting files" arch=amd64 backend=filesystem command=create container=d7f40e2e89032e5121af6496da439c11228e6f52d214bf1f5dc1f581ce811539 name=kata-runtime path=/var/lib/vc/sbs/d7f40e2e89032e5121af6496da439c11228e6f52d214bf1f5dc1f581ce811539 pid=3888 source=virtcontainers/store subsystem=store
Apr 02 21:36:20 crio-k8s kata-runtime[3888]: time="2019-04-02T21:36:20.895384579Z" level=debug msg="Deleting files" arch=amd64 backend=filesystem command=create container=d7f40e2e89032e5121af6496da439c11228e6f52d214bf1f5dc1f581ce811539 name=kata-runtime path=/run/vc/sbs/d7f40e2e89032e5121af6496da439c11228e6f52d214bf1f5dc1f581ce811539 pid=3888 source=virtcontainers/store subsystem=store
Apr 02 21:36:20 crio-k8s kata-runtime[3888]: time="2019-04-02T21:36:20.89551318Z" level=error msg="Failed to check if grpc server is working: rpc error: code = Unavailable desc = transport is closing" arch=amd64 command=create container=d7f40e2e89032e5121af6496da439c11228e6f52d214bf1f5dc1f581ce811539 name=kata-runtime pid=3888 source=runtime
Apr 02 21:36:20 crio-k8s kata-runtime[3950]: time="2019-04-02T21:36:20.928607629Z" level=info arch=amd64 arguments="\"delete --force d7f40e2e89032e5121af6496da439c11228e6f52d214bf1f5dc1f581ce811539\"" command=delete commit=c89eb81dec542ea687864d2eaabfb9c3e80dc8b5 name=kata-runtime pid=3950 source=runtime version=1.6.0-rc2
Apr 02 21:36:20 crio-k8s kata-runtime[3950]: time="2019-04-02T21:36:20.928683329Z" level=warning msg="Failed to get container, force will not fail: Container ID (d7f40e2e89032e5121af6496da439c11228e6f52d214bf1f5dc1f581ce811539) does not exist" arch=amd64 command=delete container=d7f40e2e89032e5121af6496da439c11228e6f52d214bf1f5dc1f581ce811539 name=kata-runtime pid=3950 source=runtime

```

**Comment by chavafg** [comment]
In addition to the above failures, now that we moved to k8s 1.15, found these other tests failing:
```
• Failure [90.491 seconds]
[sig-storage] EmptyDir volumes
/workspace/anago-v1.15.1-beta.0.48+4485c6f18cee9a/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/common/empty_dir.go:41
  volume on default medium should have the correct mode [LinuxOnly] [NodeConformance] [Conformance] [It]
  /workspace/anago-v1.15.1-beta.0.48+4485c6f18cee9a/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:697

  Unexpected error:
      <*errors.errorString | 0xc002c516c0>: {
          s: "expected \"perms of file \\\"/test-volume\\\": -rwxrwxrwx\" in container output: Expected\n    <string>: mount type of \"/test-volume\": 2035054128\n    perms of file \"/test-volume\": -rwxr-xr-x\n    \nto contain substring\n    <string>: perms of file \"/test-volume\": -rwxrwxrwx",
      }
      expected "perms of file \"/test-volume\": -rwxrwxrwx" in container output: Expected
          <string>: mount type of "/test-volume": 2035054128
          perms of file "/test-volume": -rwxr-xr-x
          
      to contain substring
          <string>: perms of file "/test-volume": -rwxrwxrwx
  occurred

  /workspace/anago-v1.15.1-beta.0.48+4485c6f18cee9a/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/util.go:2342
```
```
• Failure [72.199 seconds]
[sig-storage] EmptyDir volumes
/workspace/anago-v1.15.1-beta.0.48+4485c6f18cee9a/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/common/empty_dir.go:41
  should support (non-root,0644,default) [LinuxOnly] [NodeConformance] [Conformance] [It]
  /workspace/anago-v1.15.1-beta.0.48+4485c6f18cee9a/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:697

  Unexpected error:
      <*errors.errorString | 0xc003b97be0>: {
          s: "expected pod \"pod-c436daa9-c64c-4068-aec4-b4a0d5a56e0c\" success: pod \"pod-c436daa9-c64c-4068-aec4-b4a0d5a56e0c\" failed with status: {Phase:Failed Conditions:[{Type:Initialized Status:True LastP
robeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2019-08-28 15:40:20 +0000 UTC Reason: Message:} {Type:Ready Status:False LastProbeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2019-08-28 15:40:
20 +0000 UTC Reason:ContainersNotReady Message:containers with unready status: [test-container]} {Type:ContainersReady Status:False LastProbeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2019-08-28 15:40
:20 +0000 UTC Reason:ContainersNotReady Message:containers with unready status: [test-container]} {Type:PodScheduled Status:True LastProbeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2019-08-28 15:40:20
 +0000 UTC Reason: Message:}] Message: Reason: NominatedNodeName: HostIP:10.0.0.4 PodIP:10.244.0.51 StartTime:2019-08-28 15:40:20 +0000 UTC InitContainerStatuses:[] ContainerStatuses:[{Name:test-container State:
{Waiting:nil Running:nil Terminated:&ContainerStateTerminated{ExitCode:1,Signal:0,Reason:Error,Message:,StartedAt:2019-08-28 15:40:24 +0000 UTC,FinishedAt:2019-08-28 15:40:24 +0000 UTC,ContainerID:cri-o://3c45ba
3b699ab6e9e78133330b04c6b85b84f34456a7b6b3e95138e94057bfc1,}} LastTerminationState:{Waiting:nil Running:nil Terminated:nil} Ready:false RestartCount:0 Image:gcr.io/kubernetes-e2e-test-images/mounttest-user:1.0 I
mageID:gcr.io/kubernetes-e2e-test-images/mounttest-user@sha256:1ac490f1442cad4b1c886527470393c71302cc62f5e3304fb08854403cbc68dc ContainerID:cri-o://3c45ba3b699ab6e9e78133330b04c6b85b84f34456a7b6b3e95138e94057bfc
1}] QOSClass:BestEffort}",
      }
      expected pod "pod-c436daa9-c64c-4068-aec4-b4a0d5a56e0c" success: pod "pod-c436daa9-c64c-4068-aec4-b4a0d5a56e0c" failed with status: {Phase:Failed Conditions:[{Type:Initialized Status:True LastP
robeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2019-08-28 15:40:20 +0000 UTC Reason: Message:} {Type:Ready Status:False LastProbeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2019-08-28 15:40:
20 +0000 UTC Reason:ContainersNotReady Message:containers with unready status: [test-container]} {Type:ContainersReady Status:False LastProbeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2019-08-28 15:40
:20 +0000 UTC Reason:ContainersNotReady Message:containers with unready status: [test-container]} {Type:PodScheduled Status:True LastProbeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2019-08-28 15:40:20
 +0000 UTC Reason: Message:}] Message: Reason: NominatedNodeName: HostIP:10.0.0.4 PodIP:10.244.0.51 StartTime:2019-08-28 15:40:20 +0000 UTC InitContainerStatuses:[] ContainerStatuses:[{Name:test-container State:
{Waiting:nil Running:nil Terminated:&ContainerStateTerminated{ExitCode:1,Signal:0,Reason:Error,Message:,StartedAt:2019-08-28 15:40:24 +0000 UTC,FinishedAt:2019-08-28 15:40:24 +0000 UTC,ContainerID:cri-o://3c45ba
3b699ab6e9e78133330b04c6b85b84f34456a7b6b3e95138e94057bfc1,}} LastTerminationState:{Waiting:nil Running:nil Terminated:nil} Ready:false RestartCount:0 Image:gcr.io/kubernetes-e2e-test-images/mounttest-user:1.0 I
mageID:gcr.io/kubernetes-e2e-test-images/mounttest-user@sha256:1ac490f1442cad4b1c886527470393c71302cc62f5e3304fb08854403cbc68dc ContainerID:cri-o://3c45ba3b699ab6e9e78133330b04c6b85b84f34456a7b6b3e95138e94057bfc
1}] QOSClass:BestEffort}
  occurred

  /workspace/anago-v1.15.1-beta.0.48+4485c6f18cee9a/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/util.go:2342

```

Taking a look at the 1st failure, seems that maybe kata is returning output with incorrect format, but need to investigate more. 
/cc @egernst @amshinde

**Comment by amshinde** [comment]
@chavafg I had run the tests prior to the holidays with crio 1.15.1 and kubernetes 1.15.7. 
These were the failures that I was seeing:
```
Summarizing 16 Failures:

[Fail] [sig-storage] Downward API volume [It] should provide container's cpu request [NodeConformance] [Conformance]
/workspace/anago-v1.15.7-beta.0.34+76e131a00b86a9/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/util.go:2342

[Fail] [sig-storage] Downward API volume [It] should provide container's memory request [NodeConformance] [Conformance]
/workspace/anago-v1.15.7-beta.0.34+76e131a00b86a9/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/util.go:2342

[Fail] [k8s.io] InitContainer [NodeConformance] [It] should invoke init containers on a RestartAlways pod [Conformance]
/workspace/anago-v1.15.7-beta.0.34+76e131a00b86a9/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/common/init_container.go:170

[Fail] [sig-storage] Projected downwardAPI [It] should provide container's memory limit [NodeConformance] [Conformance]
/workspace/anago-v1.15.7-beta.0.34+76e131a00b86a9/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/util.go:2342

[Fail] [sig-node] Downward API [It] should provide container's limits.cpu/memory and requests.cpu/memory as env vars [NodeConformance] [Conformance]
/workspace/anago-v1.15.7-beta.0.34+76e131a00b86a9/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/util.go:2342

[Fail] [sig-storage] Projected downwardAPI [It] should provide container's cpu request [NodeConformance] [Conformance]
/workspace/anago-v1.15.7-beta.0.34+76e131a00b86a9/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/util.go:2342

[Fail] [k8s.io] InitContainer [NodeConformance] [It] should not start app containers if init containers fail on a RestartAlways pod [Conformance]
/workspace/anago-v1.15.7-beta.0.34+76e131a00b86a9/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/common/init_container.go:292

[Fail] [k8s.io] InitContainer [NodeConformance] [It] should not start app containers and fail the pod if init containers fail on a RestartNever pod [Conformance]
/workspace/anago-v1.15.7-beta.0.34+76e131a00b86a9/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/common/init_container.go:401

[Fail] [sig-storage] Projected downwardAPI [It] should provide container's cpu limit [NodeConformance] [Conformance]
/workspace/anago-v1.15.7-beta.0.34+76e131a00b86a9/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/util.go:2342

[Fail] [sig-node] Downward API [It] should provide host IP as an env var [NodeConformance] [Conformance]
/workspace/anago-v1.15.7-beta.0.34+76e131a00b86a9/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/util.go:2342

[Fail] [sig-storage] Projected downwardAPI [It] should provide container's memory request [NodeConformance] [Conformance]
/workspace/anago-v1.15.7-beta.0.34+76e131a00b86a9/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/util.go:2342

[Fail] [sig-storage] Downward API volume [It] should provide container's memory limit [NodeConformance] [Conformance]
/workspace/anago-v1.15.7-beta.0.34+76e131a00b86a9/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/util.go:2342

[Fail] [sig-node] Downward API [It] should provide pod name, namespace and IP address as env vars [NodeConformance] [Conformance]
/workspace/anago-v1.15.7-beta.0.34+76e131a00b86a9/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/util.go:2342

[Fail] [sig-storage] Downward API volume [It] should provide container's cpu limit [NodeConformance] [Conformance]
/workspace/anago-v1.15.7-beta.0.34+76e131a00b86a9/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/util.go:2342

[Fail] [sig-apps] Daemon set [Serial] [It] should rollback without unnecessary restarts [Conformance]
/workspace/anago-v1.15.7-beta.0.34+76e131a00b86a9/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/apps/daemon_set.go:386

[Fail] [sig-node] Downward API [It] should provide pod UID as env vars [NodeConformance] [Conformance]
/workspace/anago-v1.15.7-beta.0.34+76e131a00b86a9/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/util.go:2342
```

On analyzing the failures and looking at the tests itself, I realised that the failures seen are mostly caused due to the resource constraints placed on the containers, which are insufficient for launching a Kata container.
All the ` Downward API` , `Projected downwardAPI` and the `Downward API volume` tests are failing due to this.
These tests have resource constraints as :
https://github.com/kubernetes/kubernetes/blob/f1e912c38abcecfb64e36eb161128b15e69a135b/test/e2e/common/downwardapi_volume.go#L334

On increasing the CPU and memory constraints, I was able verify the test pass for a manual pod creation.

Same goes for the `InitContainer` tests. The failing tests have a CPU limit as seen here:
https://github.com/kubernetes/kubernetes/blob/4e35750abcfc7848719b84dd04346488fc2aa9fa/test/e2e/common/init_container.go#L262

I verified tests pass manually with kata.

Finally, the last test that fails is 
```[Fail] [sig-apps] Daemon set [Serial] [It] should rollback without unnecessary restarts [Conformance]
/workspace/anago-v1.15.7-beta.0.34+76e131a00b86a9/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/apps/daemon_set.go:386
```
From the logs, this test fails with the following error:
```
Conformance test suite needs a cluster with at least 2 nodes.
```
The test was run on a cluster with just 1 node, we need to test this on a cluster with atleast 2 nodes.
I havent got to that yet.

So in summary,  we should run the tests adding in the pod overhead.
Today we have a controller for making the workload run with Kata:
https://github.com/kata-containers/tests/tree/master/kata-webhook
I suppose we need another admission controller for the pod-overhead
@egernst can you add your input on this.

**Comment by egernst** [comment]
Nice work @amshinde.  PodOverhead is only available, in beta, starting in 1.16.  

We can't rely on it here. Perhaps we can re-run on 1.16 with the feature gates set appropriately.

**Comment by chavafg** [comment]
thanks @amshinde for finding the root cause. 
Can you confirm that the `EmptyDir volumes` tests passed on your side. These were also failing when I ran them.

**Comment by amshinde** [comment]
@chavafg I didnt see any failures for `EmptyDir volumes` when I ran the tests. I'll try another run with 1.16 today.

**Comment by chavafg** [comment]
@amshinde 
I used PodOverhead as you recommended and the failing tests now worked correctly. 
I only saw 2 failures now:

1. one `Downward API ` test failed with an strange `etcd` timeout issue. seems not related to the same failure as before, will check furtherly
```
• Failure [61.972 seconds]
[sig-node] Downward API
/workspace/anago-v1.16.2-beta.0.19+c97fe5036ef3df/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/common/downward_api.go:32
  should provide pod UID as env vars [NodeConformance] [Conformance] [It]
  /workspace/anago-v1.16.2-beta.0.19+c97fe5036ef3df/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:698

  Jan 17 17:32:23.180: Failed to delete pod "downward-api-2a869573-3f20-4641-b4e5-fcd973c3c8eb": etcdserver: request timed out

  /workspace/anago-v1.16.2-beta.0.19+c97fe5036ef3df/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/pods.go:175
```

2. I still saw an issue with the `DaemonSet` test, which requires 2 nodes at least. I think that we can skip that for the moment, until we find a way to launch a 2 node cluster. 
```
[sig-apps] Daemon set [Serial]
/workspace/anago-v1.16.2-beta.0.19+c97fe5036ef3df/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/apps/framework.go:23
  should rollback without unnecessary restarts [Conformance] [It]
  /workspace/anago-v1.16.2-beta.0.19+c97fe5036ef3df/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:698

  Jan 17 19:33:44.035: Conformance test suite needs a cluster with at least 2 nodes.
  Expected
      <int>: 1
  to be >
      <int>: 1
```

I will update our CI systems to use PodOverhead.

**Comment by amshinde** [comment]
@mrunalp As we discussed, we now have the e2e tests passing with sonobuoy. cc @chavafg 
Let us know if you have any other tests in mind.

**Comment by amshinde** [comment]
@mrunalp So we run the e2e tests nightly. 
@chavafg Can you post the link to access the nightly build results?

**Comment by chavafg** [comment]
here you can see the nightly jobs running...

http://jenkins.katacontainers.io/job/kata-containers-crio-k8s-conformance-e2e/ 

we are currently skipping one [`Daemon Set`](https://github.com/kata-containers/tests/blob/master/integration/kubernetes/e2e_conformance/skipped_tests_e2e.yaml#L14) test because it requires a 2 node cluster and we are using only one.

**Comment by haircommander** [comment]
I believe we can consider this done. WDYT @amshinde @egernst @chavafg

**Comment by fidencio** [comment]
I've added part of this on kata-containers side but it's not running daily / weekly.
This is something I think either @fgiudici or @littlejawa will work on / maintain.  Francesco, Julien, could you confirm or deny that? :-)

**Comment by littlejawa** [comment]
Right :)
Our first goal is to run it on a regular basis (but on a smaller set of tests).
Then increase the coverage by fixing anything that prevents the tests from running.

The second goal is obviously longer term, but we're working on it.
